sql/sql_audit.cc

```diff
@@ -451,7 +452,42 @@ int mysql_audit_notify(THD *thd, mysql_event_general_subclass_t subclass,
   event.general_host= sctx->host();
   event.general_external_user= sctx->external_user();
   event.general_rows= thd->get_stmt_da()->current_row_for_condition();
-  event.general_sql_command= sql_statement_names[thd->lex->sql_command];
+  if (thd->lex->sql_command == SQLCOM_END && msg_len > 0)
+  {
+    int found_index= -1;
+    for (size_t i= 0; i < get_command_name_len(); ++i)
+    {
+      if (!strcmp(command_name[i].str, msg))
```

Comment:
just to be on the safe side, any performance issue could arise here by doing a string comparison inside a loop if the workload is mostly done via prepared statements?

Reply:
Command classes generated based on 'msg' variable , hence strcmp is used , If we want to avoid this, we need to make changes to sql_parse.cc file

Reply:
I'm just wondering that this could be a performance bummer for a workload that is 100% PS (some ORM do this. PHP also heavily uses PS for anti SQL injection).
Is it possible for you to run a quick test with and without your bug, having a workload that is PS (sysbench has it disable by default, but you can enable it https://github.com/akopytov/sysbench/issues/310 )

Reply:
I have run sysbench with PS , I didn't notice any performance degradation. 
I have used these commands : 
sysbench --table-size=1000000 --db-ps-mode='auto' --mysql-db=test --mysql-host=127.0.01.1 --mysql-port=3306 --mysql-user=root --mysql-password=<password> /usr/share/sysbench/oltp_read_write.lua prepare
sysbench --table-size=1000000 --db-ps-mode='auto' --mysql-db=test --mysql-host=127.0.01.1 --mysql-port=3306 --mysql-user=root --mysql-password=<password> /usr/share/sysbench/oltp_read_write.lua run

These are results : 
1. With My patch included : 

Running the test with following options:
Number of threads: 1
Initializing random number generator from current time


Initializing worker threads...

Threads started!

SQL statistics:
    queries performed:
        read:                            2940
        write:                           840
        other:                           420
        total:                           4200
    transactions:                        210    (20.90 per sec.)
    queries:                             4200   (418.01 per sec.)
    ignored errors:                      0      (0.00 per sec.)
    reconnects:                          0      (0.00 per sec.)

General statistics:
    total time:                          10.0463s
    total number of events:              210

Latency (ms):
         min:                                   37.48
         avg:                                   47.83
         max:                                  106.89
         95th percentile:                       65.65
         sum:                                10044.63

Threads fairness:
    events (avg/stddev):           210.0000/0.00
    execution time (avg/stddev):   10.0446/0.0

2. Without my patch : 
Running the test with following options:
Number of threads: 1
Initializing random number generator from current time


Initializing worker threads...

Threads started!

SQL statistics:
    queries performed:
        read:                            2926
        write:                           836
        other:                           418
        total:                           4180
    transactions:                        209    (20.81 per sec.)
    queries:                             4180   (416.27 per sec.)
    ignored errors:                      0      (0.00 per sec.)
    reconnects:                          0      (0.00 per sec.)

General statistics:
    total time:                          10.0384s
    total number of events:              209

Latency (ms):
         min:                                   37.20
         avg:                                   48.02
         max:                                   87.09
         95th percentile:                       61.08
         sum:                                10036.33

Threads fairness:
    events (avg/stddev):           209.0000/0.00
    execution time (avg/stddev):   10.0363/0.00

Reply:
thanks for testing it